### PR TITLE
[ADD] Support convolutions with string-valued padding

### LINKED ISSUE #35: Support convolutions with string-valued padding
(follow-up from #34)

The current version of `_extract_patches` does not support string-valued padding, but only 2-tuples, although strings are allowed to specify the padding in `torch.nn.Conv2d`.

We could use [this](https://github.com/f-dangel/einconv/blob/master/einconv/utils.py#L97-L132) function from my `einconv` library to compute the left and right paddings given the convolution's hyper-parameters.

### PR DESCRIPTION
**Pull request by f-dangel**
Resolves #35. Only review after merging #51.

### CODE REVIEW

**Review by runame — APPROVED** [review]
Haven't checked what `get_conv_paddings` exactly does, but otherwise LGTM.
